### ISSUE TITLE
Can't format long comments, fails with 3

### ISSUE THREAD

**Issue by LeoTindall** [rust-lang-nursery/rustfmt#1852]
I have a file with the following contents.

```
/// This is a sample Rust file with a line that is too long, but is a comment. This really should be formattable, but apparently isn't.
type Something = i32;

// This line is much the same. It's not a doc comment, though, which the other one was. In any case, they should both be formattable.
```

Running rustfmt on it results in status 3 and two warnings about line length. Shouldn't it be possible to 
just wrap the comments?
```
leo@Shadowmere /tmp> rustfmt --version
0.9.0-nightly ( )
leo@Shadowmere /tmp> rustfmt test.rs; echo $status
Rustfmt failed at /tmp/test.rs:1: line exceeded maximum length (maximum: 100, found: 135) (sorry)
Rustfmt failed at /tmp/test.rs:4: line exceeded maximum length (maximum: 100, found: 133) (sorry)

3
```

**Comment by Majora320** [comment]
Possible fix:
By default, warn (not fail) on too-long comment lines. Provide an option to split too-long comment lines by word and reformat.

**Comment by topecongiro** [comment]
You could use `wrap_comments = true` to format comments, although it is known to be fragile.
Alternatively, you could use `error_on_line_overflow_comments = false` to suppress warnings against single line comments. This option is only available to `rustfmt-nightly`.

**Comment by topecongiro** [comment]
Closed via #2257.
